barretenberg/cpp/src/barretenberg/crypto/merkle_tree/node_store/cached_content_addressed_tree_store.hpp

```diff
@@ -259,9 +254,7 @@ template <typename LeafValueType> class ContentAddressedCachedTreeStore {
 
     void remove_leaf(const fr& hash, const std::optional<index_t>& maxIndex, WriteTransaction& tx);
 
-    void remove_leaf_indices(const fr& key, const index_t& maxIndex, WriteTransaction& tx);
-
-    void remove_leaf_indices_after_or_equal_index(const index_t& maxIndex, WriteTransaction& tx);
+    void remove_leaf_index(const fr& key, const index_t& maxIndex, WriteTransaction& tx);
```

Comment:
Same question here: could we remove the `maxIndex`? What is it used for?

Reply:
Where you see `maxIndex`, we are executing the function from the point of view of a specific block number. So in the case of a read query we limit the results returned to include those leaves present at the time that block was included. 

In the case of a write transaction (an unwind specifically) we only delete the leaf if it was included after the new pending chian tip. It might not seem obvious why we need to check this. When a public data value is updated, it's value changes producing a new leaf node hash. The leaf index however is based on slot number which does not change. If you unwind a block from the public data tree, the tree search will return leaf nodes at the indices that the slot was originally inserted, which means that slot should not be removed from the index store unless the index is beyond the new pending chain (i.e. the block you are unwinding introduced the slot for the first time).